### PR TITLE
Adding commons-io 2.4

### PR DESCRIPTION
**Pull request by pulasthi7**

### CODE REVIEW

File: bcpkix/1.49.0.wso2v2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                            org.bouncycastle.x509.*;version="[1.49.0.wso2v2, 1.50.0.wso2v1)",
+                            org.bouncycastle.util.*;version="[1.49.0.wso2v2, 1.50.0.wso2v1)",
+                            org.bouncycastle.pqc.*;version="[1.49.0.wso2v2, 1.50.0.wso2v1)",
+                            org.bouncycastle.ocsp.*;version="[1.49.0.wso2v2, 1.50.0.wso2v1)",
+                            org.bouncycastle.math.*;version="[1.49.0.wso2v2, 1.50.0.wso2v1)",
+                            org.bouncycastle.jce.*;version="[1.49.0.wso2v2, 1.50.0.wso2v1)",
+                            org.bouncycastle.jcajce.*;version="[1.49.0.wso2v2, 1.50.0.wso2v1)",
+                            org.bouncycastle.i18n.*;version="[1.49.0.wso2v2, 1.50.0.wso2v1)",
+                            org.bouncycastle.crypto.*;version="[1.49.0.wso2v2, 1.50.0.wso2v1)",
+                            org.bouncycastle.asn1.*;version="[1.49.0.wso2v2, 1.50.0.wso2v1)"
```
Comment (Kishanthan) [inline bcpkix/1.49.0.wso2v2/pom.xml]: Use a property for the version range and refer that here.

File: bcpkix/1.49.0.wso2v2/pom.xml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                        </Import-Package>
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
Comment (Kishanthan) [inline bcpkix/1.49.0.wso2v2/pom.xml]: What is the reason to include Dynamic-Import package * here?

File: jscep/2.0.2.wso2v2/pom.xml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                            org.jscep.*;version="2.0.2",
+                        </Export-Package>
+                        <Import-Package>
+                            org.apache.commons.io.*;version="[2.4.0, 2.5.0)",
+                            !org.jscep.*,
+                            *;resolution:=optional
+                        </Import-Package>
+                        <Private-Package>
+                        </Private-Package>
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
Comment (Kishanthan) [inline jscep/2.0.2.wso2v2/pom.xml]: What is the reason to include Dynamic-Import package * here?

File: jscep/2.0.2.wso2v2/pom.xml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                        <Export-Package>
+                            org.jscep.*;version="2.0.2",
+                        </Export-Package>
+                        <Import-Package>
+                            org.apache.commons.io.*;version="[2.4.0, 2.5.0)",
+                            !org.jscep.*,
+                            *;resolution:=optional
+                        </Import-Package>
+                        <Private-Package>
+                        </Private-Package>
```
Comment (Kishanthan) [inline jscep/2.0.2.wso2v2/pom.xml]: Remove empty elements

File: jscep/2.0.2.wso2v2/pom.xml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>2.4</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.jscep</groupId>
+            <artifactId>jscep</artifactId>
+            <version>2.0.2</version>
```
Comment (Kishanthan) [inline jscep/2.0.2.wso2v2/pom.xml]: Mark this as optional dependency

File: jscep/2.0.2.wso2v2/pom.xml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                <checksumPolicy>fail</checksumPolicy>
+            </releases>
+        </repository>
+    </repositories>
+
+    <dependencies>
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>2.4</version>
```
Comment (Kishanthan) [inline jscep/2.0.2.wso2v2/pom.xml]: Why do we need this dependency here?

File: commons-io/2.4.0.wso2v1/pom.xml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                <checksumPolicy>fail</checksumPolicy>
+            </releases>
+        </repository>
+    </repositories>
+
+    <dependencies>
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>2.4</version>
```
Comment (Kishanthan) [inline commons-io/2.4.0.wso2v1/pom.xml]: Mark this as a optional dependency

File: bcpkix/1.49.0.wso2v2/pom.xml [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+                        </_exportcontents>
+                    </instructions>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <properties>
+        <exp.pkg.version.bcpkix>1.49.0.wso2v2</exp.pkg.version.bcpkix>
+        <imp.pkg.version.range.bcprov>[1.49.0.wso2v2, 1.50.0.wso2v1)</imp.pkg.version.range.bcprov>
```
Comment (Kishanthan) [inline bcpkix/1.49.0.wso2v2/pom.xml]: This should be [1.49.0.wso2v2, 1.50.0)

File: jscep/2.0.2.wso2v2/pom.xml [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+                <checksumPolicy>fail</checksumPolicy>
+            </releases>
+        </repository>
+    </repositories>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.google.code.jscep</groupId>
+            <artifactId>jscep</artifactId>
+            <version>2.0.2</version>
```
Comment (Kishanthan) [inline jscep/2.0.2.wso2v2/pom.xml]: Can we make this an optional dependency?

Comment (pulasthi7) [inline jscep/2.0.2.wso2v2/pom.xml]: Fixed.

File: commons-io/2.4.0.wso2v1/pom.xml (hunk 7 above)
Comment (pulasthi7) [inline commons-io/2.4.0.wso2v1/pom.xml]: Fixed.

**Pull request merged**